Psychtoolbox/PsychHardware/EyelinkToolbox/EyelinkDemos/SR-ResearchDemos/GazeContingent/FixWindowBufferedSamples/EyeLink_FixWindowBufferedSamples.m

```diff
@@ -0,0 +1,538 @@
+%
+% Use buffered data if you need to:
+% a) grab every single consecutive sample online
+% b) grab event data (e.g. fixation/saccade/blink events) online
+
+% Note that some buffered event data take some time to be available online due to the times involved
+% in calculating velocity/acceleration. If you need to retrieve online gaze
+% position as fast as possible and/or you don't need to get all subsequent samples or other
+% events, then use the Eyelink('NewFloatSampleAvailable') / Eyelink('NewestFloatSample') function pair,
+% as illustrated in the GCfastSamples.m example.
+% ---------------------------------------------------------------------------------------------
+%
+% Events structure and fields available via the 'GetNextDataType'/'GetFloatData' function pair:
+% STARTBLINK, STARTSACC, STARTFIX:
+%       type (number assigned to event - STARTBLINK=3, STARTSACC=5, STARTFIX=7)
+%       eye (0=left eye, 1=right eye)
+%       sttime (event start time)
+%
+% ENDBLINK:
+%       type (number assigned to event - ENDBLINK=4)
+%       eye (0=left eye, 1=right eye)
+%       sttime (event start time)
+%       entime (event end time)
+%
+% ENDSACC:
+%       type (number assigned to event - ENDSACC=6)
+%       eye (0=left eye, 1=right eye)
+%       sttime (event start time)
+%       entime (event end time)
+%       gstx (Saccade start x gaze position)
+%       gsty (Saccade start y gaze position)
+%       genx (Saccade end x gaze position)
+%       geny (Saccade end y gaze position)
+%       supd_x (Saccade start x 'pixel per degree' value)
+%       supd_y (Saccade start y 'pixel per degree' value)
+%       eupd_x (Saccade end x 'pixel per degree' value)
+%       eupd_y (Saccade end y 'pixel per degree' value)
+%
+% FIXUPDATE, ENDFIX:
+%       type (number assigned to event - FIXUPDATE=9, ENDFIX=8)
+%       eye (0=left eye, 1=right eye)
+%       sttime (event start time)
+%       entime (event end time)
+%       gavx (average gaze x position during fixation)
+%       gavy (average gaze y position during fixation)
+%       ava (average pupil size)
+%       supd_x (Fixation start x 'pixel per degree' value)
+%       supd_y (Fixation start y 'pixel per degree' value)
+%       eupd_x (Fixation end x 'pixel per degree' value)
+%       eupd_y (Fixation end y 'pixel per degree' value)
+%
+% SAMPLE_TYPE
+%       time (sample time)
+%       type (SAMPLE=200)
+%       pa ([lef eye pupil size, right eye pupil size])
+%       gx ([left gaze x, right gaze x])
+%       gy ([left gaze y, right gaze y])
+%       rx (x 'pixel per degree' value)
+%       ry (y 'pixel per degree' value)
+%       buttons (button state and changes)
+%       hdata (contains a list of 8 fields. Only the first 4 values are important:
+%             [uncalibrated target sticker x, uncalibrated target sticker y, target sticker distance in mm, target flags)
+
+% Bring the Command Window to the front if it is already open
+if ~IsOctave; commandwindow; end
+

```

Comment:
``PsychDefaultSetup()`` is missing. Recommended is at least ``PsychDefaultSetup(1)`` for unified key mappings across operating systems, better ``PsychDefaultSetup(2)`` in combination with ``PsychImaging('Openwindow', ...)`` for a normalized color range from 0 - 1 for SDR displays. Obviously all colors and colors in double() textures must then be specified in normalized range. Simple uint8() texture images, e.g., read via imread() etc. are not affected.

Same probably applies to all other demos.

Reply:
resolved in cf762efe7df1a9e7d29650dfcb83c66c0b571bd7